### PR TITLE
add LinearAlgebraExt

### PR DESCRIPTION
**Pull request by longemen3000**
should make the package dependency free on julia > 1.9. motivated by [JuliaFolds2/Trand](https://github.com/JuliaFolds2/Transducers.jl/issues/35)

### CODE REVIEW

**Comment by jw3126** [comment]
Hey thanks. In principle I agree this is a good move. However we were burnt by something similar before https://github.com/JuliaObjects/Accessors.jl/issues/127

@aplavin you think this PR would trigger the same issues?

**Comment by longemen3000** [comment]
seems that the problem is (somewhat) solved in 1.11, so maybe a good idea would be to guard extensions against that release instead?

File: test/runtests.jl [hunk 1]
```diff
@@ -21,7 +21,15 @@ end
...
     @test getfields(()) === ()
-    @test getfields([]) === NamedTuple()
+
+    #on julia 1.10 onwards, Array has fields :ref and :size. the :ref field is a memory field
+    #with non-constant value (the pointer location in memory). The only constant field in []
+    #is it's size, (0,)
+    if !isdefined(Base,:Memory)
+        @test getfields([]) === NamedTuple()
+    else
+        @test getfields([]).size === (0,)
```
Comment (jw3126) [inline test/runtests.jl:31]:
```suggestion
        arr = []
        @test getfields(arr) === (;arr.ref, arr.size)
```

Comment (jw3126) [reply test/runtests.jl:31]: could also test for empty getproperties, but maybe we already do elsewhere.

**Comment by jw3126** [comment]
> seems that the problem is (somewhat) solved in 1.11, so maybe a good idea would be to guard extensions against that release instead? (see [JuliaLang/julia#52511](https://github.com/JuliaLang/julia/issues/52511))

How would that work? Would you add `julia = "1.11"` to `ConstructionBase/Project.toml`?

**Comment by longemen3000** [comment]
> How would that work? Would you add ``

i was thinking on just guarding the loading of extensions against `if Base.VERSION < v"1.11"` instead of using `if !isdefined(Base,:get_extension)`

**Comment by rafaqz** [comment]
Guarding against 1.11 seems fine to me, I guess you won't be using the lts in projects that need to trim LinearAlgebra.jl anyway.

**Comment by longemen3000** [comment]
What about just having the structure ready for when the problem is solved?, I suppose that this problem will be solved in a future 1.10 release and a 1.11 one, so it is just a matter of waiting for those releases, and loading the ext unconditionally until then?

**Comment by longemen3000** [comment]
i would say that this PR is ready to merge. as mentioned before, this PR no longer sets `LinearAlgebra` as an extension, but it paves the way to allow that change.

test errors are integration errors, that are also present on the master branch

**Review by jw3126 — APPROVED** [review]
LGTM, thanks @longemen3000
